protected/models/PatientUserReferral.php

```diff
@@ -0,0 +1,125 @@
+			array('last_modified_date, created_date', 'safe'),
+			// The following rule is used by search().
+			// @todo Please remove those attributes that should not be searched.
+			array('id, patient_id, user_id, last_modified_user_id, last_modified_date, created_user_id, created_date', 'safe', 'on'=>'search'),
+		);
+	}
+
+	/**
+	 * @return array relational rules.
+	 */
+	public function relations()
+	{
+		// NOTE: you may need to adjust the relation name and the related
+		// class name for the relations automatically generated below.
+		return array(
+			'user' => array(self::BELONGS_TO, 'User', 'user_id'),
+			'createdUser' => array(self::BELONGS_TO, 'User', 'created_user_id'),
+			'lastModifiedUser' => array(self::BELONGS_TO, 'User', 'last_modified_user_id'),
+			'patient' => array(self::BELONGS_TO, 'Patient', 'patient_id'),
+		);
+	}
+
+	/**
+	 * @return array customized attribute labels (name=>label)
+	 */
+	public function attributeLabels()
+	{
+		return array(
+			'id' => 'ID',
+			'patient_id' => 'Patient',
+			'user_id' => 'User',
+			'last_modified_user_id' => 'Last Modified User',
+			'last_modified_date' => 'Last Modified Date',
+			'created_user_id' => 'Created User',
+			'created_date' => 'Created Date',
+		);
+	}
+
+	/**
+	 * Retrieves a list of models based on the current search/filter conditions.
+	 *
+	 * Typical usecase:
+	 * - Initialize the model fields with values from filter form.
+	 * - Execute this method to get CActiveDataProvider instance which will filter
+	 * models according to data in model fields.
+	 * - Pass data provider to CGridView, CListView or any similar widget.
+	 *
+	 * @return CActiveDataProvider the data provider that can return the models
+	 * based on the search/filter conditions.
+	 */
+	public function search()
+	{
+		// @todo Please modify the following code to remove attributes that should not be searched.
+
+		$criteria=new CDbCriteria;
+
+		$criteria->compare('id',$this->id);
+		$criteria->compare('patient_id',$this->patient_id,true);
+		$criteria->compare('user_id',$this->user_id,true);
+		$criteria->compare('last_modified_user_id',$this->last_modified_user_id,true);
+		$criteria->compare('last_modified_date',$this->last_modified_date,true);
+		$criteria->compare('created_user_id',$this->created_user_id,true);
+		$criteria->compare('created_date',$this->created_date,true);
+
+		return new CActiveDataProvider($this, array(
+			'criteria'=>$criteria,
+		));
+	}
+
+	/**
+	 * Returns the static model of the specified AR class.
+	 * Please note that you should have this exact method in all your CActiveRecord descendants!
+	 * @param string $className active record class name.
+	 * @return PatientUserReferral the static model class
+	 */
+	public static function model($className=__CLASS__)
+	{
+		return parent::model($className);
+	}
+
+    public function getUserName()
+    {
+        return $this->user->getFullName();
+    }
+}
```

Comment:
Something seems a bit off with the alignment of the functions in this class. CTRL-Alt-L should fix this.